library/src/window_impl.cxx

```diff
@@ -72,17 +72,30 @@ class window_impl::internals
     // Override VTK logic
 #ifdef _WIN32
     return vtkSmartPointer<vtkF3DWGLRenderWindow>::New();
-#else
+#elif __linux__
 #if defined(VTK_USE_X)
     // try GLX
-    vtkSmartPointer<vtkRenderWindow> renWin = vtkSmartPointer<vtkF3DGLXRenderWindow>::New();
-    if (renWin)
+    vtkSmartPointer<vtkRenderWindow> glxRenWin = vtkSmartPointer<vtkF3DGLXRenderWindow>::New();
+    if (glxRenWin)
+    {
+      return glxRenWin;
+    }
+#endif
+#if defined(VTK_OPENGL_HAS_EGL)
+    // try EGL
+    vtkSmartPointer<vtkRenderWindow> eglRenWin = vtkSmartPointer<vtkF3DEGLRenderWindow>::New();
+    if (eglRenWin)
     {
-      return renWin;
+      return eglRenWin;
     }
 #endif
-    // XXX: At the moment, fallback on VTK logic
-    // It will change in the future when other subclasses are implemented
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+    // OSMesa
+    return vtkSmartPointer<vtkOSOpenGLRenderWindow>::New();
```

Comment:
i would expect this to be covered ?

Reply:
The coverage job is running with a X server so the automatic mode will instantiate a GLX context.

Reply:
right